### DISCUSSION TITLE
Right-to-Left Language caret position issue.

### DISCUSSION THREAD

**Discussion by Navid61** [jgraph/drawio#3986] [Ideas]
This program is exceptionally excellent and practical. I just want to know how to add right-to-left language support to it, similar to the Bootstrap library, which added right-to-left language support from version 5 onwards. This allows the typed text to seamlessly continue in front of the previous words when the user switches the typing language, ensuring a smooth transition.

**Comment by davidjgraph** [comment]
What did you try and what result did you see?

**Comment by Navid61** [comment]
I started typing with a right-to-left language, where I would write an English word, then switch back to the same language instead of continuing with English words. The cursor marker was placed at both the end and beginning of the English word, indicating errors at the end of the English word and the end of the right-to-left text.

**Reply by davidjgraph** [reply]
OK, noted. However, this is the built in browser text field we're using, not sure if we can change the behaviour.

**Reply by Navid61** [reply]
Thank you. In my opinion, the draw.io library is excellent, and I would like to see support for right-to-left languages added to it to encourage more people to use it. This could be considered as an intrinsic and essential feature of the library.

**Reply by davidjgraph** [reply]
Just to clarify, there is support for right-to-left languages. The specific problem here is a claret positioning change when switching languages.
